src/vmm/src/arch/x86_64/mptable.rs

```diff
@@ -369,6 +378,39 @@ mod tests {
         );
     }
 
+    #[test]
+    fn mpc_entry_count() {
+        let num_cpus = 1;
+        let mem = GuestMemoryMmap::from_raw_regions(
+            &[(GuestAddress(MPTABLE_START), compute_mp_size(num_cpus))],
+            false,
+        )
+        .unwrap();
+
+        setup_mptable(&mem, num_cpus).unwrap();
+
+        let mpf_intel: mpspec::mpf_intel = mem.read_obj(GuestAddress(MPTABLE_START)).unwrap();
+        let mpc_offset = GuestAddress(u64::from(mpf_intel.physptr));
+        let mpc_table: mpspec::mpc_table = mem.read_obj(mpc_offset).unwrap();
+
+        let expected_entry_count = 0 as u16
+            // Intel floating point
+            + 1
+            // CPU
+            + num_cpus as u16
+            // IOAPIC
+            + 1
+            // ISA Bus
+            + 1
+            // IRQ
+            + (IRQ_MAX as u16) + 1
```

Comment:
Extra +1 is from ranging over `0..=IRQ_MAX` (inclusive) here:

https://github.com/firecracker-microvm/firecracker/blob/1afbacadfa7e9891e654928044ecfb322642c179/src/vmm/src/arch/x86_64/mptable.rs#L211